plan_delegator/src/plan_delegator.cpp

```diff
@@ -108,172 +109,202 @@ namespace plan_delegator
+        config_.planning_topic_suffix = declare_parameter<std::string>("planning_topic_suffix", config_.planning_topic_suffix);
+        config_.trajectory_planning_rate = declare_parameter<double>("trajectory_planning_rate", config_.trajectory_planning_rate); 
+        config_.max_trajectory_duration = declare_parameter<double>("trajectory_duration_threshold", config_.max_trajectory_duration);
+        config_.min_crawl_speed = declare_parameter<double>("min_speed", config_.min_crawl_speed);
+    }
 
-        // Initialize world model
-        wml_.reset(new carma_wm::WMListener());
+    carma_ros2_utils::CallbackReturn PlanDelegator::handle_on_configure(const rclcpp_lifecycle::State &)
+    {
+        // Reset config
+        config_ = Config();
 
-        
+        get_parameter<std::string>("planning_topic_prefix", config_.planning_topic_prefix);      
+        get_parameter<std::string>("planning_topic_suffix", config_.planning_topic_suffix);
+        get_parameter<double>("trajectory_planning_rate", config_.trajectory_planning_rate); 
+        get_parameter<double>("trajectory_duration_threshold", config_.max_trajectory_duration);
+        get_parameter<double>("min_speed", config_.min_crawl_speed);
 
-        pnh_.param<std::string>("planning_topic_prefix", planning_topic_prefix_, "/plugins/");        
-        pnh_.param<std::string>("planning_topic_suffix", planning_topic_suffix_, "/plan_trajectory");
-        pnh_.param<double>("trajectory_planning_rate", trajectory_planning_rate_, 10.0);
-        pnh_.param<double>("trajectory_duration_threshold", max_trajectory_duration_, 6.0);
-        pnh_.param<double>("min_speed", min_crawl_speed_, min_crawl_speed_);
+        RCLCPP_INFO_STREAM(rclcpp::get_logger("plan_delegator"),"Done loading parameters: " << config_);
 
-        traj_pub_ = nh_.advertise<cav_msgs::TrajectoryPlan>("plan_trajectory", 5);
-        plan_sub_ = nh_.subscribe("final_maneuver_plan", 5, &PlanDelegator::maneuverPlanCallback, this);
-        twist_sub_ = nh_.subscribe<geometry_msgs::TwistStamped>("current_velocity", 5,
-            [this](const geometry_msgs::TwistStampedConstPtr& twist) {this->latest_twist_ = *twist;});
-        pose_sub_ = nh_.subscribe<geometry_msgs::PoseStamped>("current_pose", 5,
-            [this](const geometry_msgs::PoseStampedConstPtr& pose) {this->latest_pose_ = *pose;});
-        guidance_state_sub_ = nh_.subscribe<cav_msgs::GuidanceState>("guidance_state", 5, &PlanDelegator::guidanceStateCallback, this);
+        traj_pub_ = create_publisher<carma_planning_msgs::msg::TrajectoryPlan>("plan_trajectory", 5);
+        plan_sub_ = create_subscription<carma_planning_msgs::msg::ManeuverPlan>("final_maneuver_plan", 5, std::bind(&PlanDelegator::maneuverPlanCallback, this, std_ph::_1));
+        twist_sub_ = create_subscription<geometry_msgs::msg::TwistStamped>("current_velocity", 5,
+            [this](geometry_msgs::msg::TwistStamped::UniquePtr twist) {this->latest_twist_ = *twist;});
```

Comment:
Just a question about this one, is this tested? Having a callback here directly, works?

Reply:
yep! it works I tested locally